File: codes/C++/fizzbuzz.cpp
#include <string>
#include <iostream>

using std::cout;
using std::cin;
using std::endl;
using std::string;

int main() {
	int end;
	int start;
	cout << "What number do you want to start on?\n";
	cin >> start;
	cout << "What number do you want to stop on?\n";
	cin >> end;
	system("cls");
	int curr = start;
	while (curr <= end) {
		string output = "";
		if(curr % 3 == 0) output += "Fizz";
		if(curr % 5 == 0) output += "Buzz";
		if(output == "") output = std::to_string(curr);
		cout << output << endl;
		curr++;
	}

	return 0;
}

File: codes/Rust/fizzbuzz.rs
fn _check(num: isize, n: isize) -> bool {
    num % n == 0
}

fn main() {
    let mut start = 1;
    let max = 100;

    while start <= max {
        match (_check(start, 3), _check(start, 5)) {
            (true, true) => println!("FizzBuzz"),
            (true, false) => println!("Fizz"),
            (false, true) => println!("Buzz"),
            _ => println!("{}", start),
        }

        start += 1;
    }
}
